GVFS/GVFS/CommandLine/RepairVerb.cs

```diff
@@ -95,7 +97,15 @@ public override void Execute()
 
             this.Output.WriteLine();
 
-            using (JsonTracer tracer = new JsonTracer(GVFSConstants.GVFSEtwProviderName, "RepairVerb"))
+            string enlistmentId = null;

```

Comment:
This is good to do, but we should stop repeating ourselves! I can see this becoming an important thing to do in more places.

I would add an `EnlistmentId` property to `GVFSEnlistment` that initializes on first call using this logic. Then replace the logic in `InProcessMount.CreateTracer()` with `enlistment.EnlistmentId`.

```
        private bool initializedIds;
        private string enlistmentId;
        private string mountId;

...

        public string EnlistmentId
        {
            get
            {
                this.InitializeIds();
                return this.enlistmentId;
            }
        }

        public string MountId
        {
            get
            {
                this.InitializeIds();
                return this.mountId;
            }
        }

...

        private void InitializeIds()
        {
            if (this.initializedIds)
            {
                return;
            }

            GitProcess git = new GitProcess(this);


            GitProcess.Result configResult = git.GetFromLocalConfig(
                                                GVFSConstants.GitConfig.EnlistmentId);
            if (!configResult.HasErrors)
            {
                this.enlistmentId = configResult.Output.Trim();
            }

            configResult = git.GetFromLocalConfig(GVFSConstants.GitConfig.MountId);
            if (!configResult.HasErrors)
            {
                this.mountId = configResult.Output.Trim();
            }

            this.initializedIds = true;
        }
```

Reply:
Yeah I agree we need this logic in one place.  We also should also use the RepoMetadata to get the `EnlistmentId` since this is where it is stored to begin with.  We could even try both places.  We should probably also detect when there is not an entry and create a new id if there is not one.  Not sure if this would only be if the call succeeded but there wasn't a value or for an error condition as well.  If there was an error and we change the EnlistmentId reporting on the enlistment would not be right but we at least have an EnlistmentId.  There is issue #398 that is for this work.